int32 left
int32 right
int32 twist
int32 priority
int32 id